### PR TITLE
tests(PasswordField): add web and native tests

### PR DESCRIPTION
**Pull request by divyanshu013**
Adds tests for `PasswordField`, mostly similar to `BaseInput`.

PR1 - https://github.com/razorpay/blade/pull/712
PR2 - this
PR3 - https://github.com/razorpay/blade/pull/716

### CODE REVIEW

File: packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.web.test.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      <PasswordField label={label} name="password" defaultValue={userPassword} />,
+    );
+
+    const input = getByLabelText(label);
+    const revealButton = getByRole('button');
+
+    expect(input).toHaveAttribute('type', 'password');
+    expect(revealButton).toHaveAccessibleName('Show password');
+
+    await user.click(revealButton);
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.web.test.tsx:199]: Should we also assert the focus? 

Like 

press.tab()
press.enter() 
revealButton.toHaveFocus()
press.enter()
revealButton.toHaveFocus()

File: packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.native.test.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    const input = getByPlaceholderText(placeholder);
+    const button = getByRole('button');
+
+    // @ts-expect-error TS typings not being picked from library
+    expect(input).toHaveProp('secureTextEntry', true);
+    getByLabelText('Show password');
+
+    fireEvent.press(button);
+
+    // @ts-expect-error TS typings not being picked from library
```
Comment (anuraghazra) [inline packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.native.test.tsx:174]: Maybe let's create an issue for this?

**Review by anuraghazra — APPROVED** [review]
LGTM. 

Feels like I'm reviewing the same exact PR for the 3rd time.. 😆 the input tests are very similar.

File: packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.native.test.tsx (hunk 2 above)
Comment (divyanshu013) [reply packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.native.test.tsx:174]: Created https://github.com/razorpay/blade/issues/720

File: packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.web.test.tsx (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Input/PasswordField/__tests__/PasswordField.web.test.tsx:202]: Good idea, added in https://github.com/razorpay/blade/pull/715/commits/20aa6d874a45a3494838b6b2dcabd860509c8e0b

**Review by chaitanyadeorukhkar — APPROVED** [review]

**Pull request merged**